tests/drivers/build_all/sensor/sensors_trigger_own.conf

```diff
@@ -43,3 +43,5 @@ CONFIG_TMP007_TRIGGER_OWN_THREAD=y
 CONFIG_VCNL4040_TRIGGER_OWN_THREAD=y
 CONFIG_WSEN_HIDS_TRIGGER_OWN_THREAD=y
 CONFIG_WSEN_TIDS_TRIGGER_OWN_THREAD=y
+CONFIG_ADT7310_TRIGGER=y
+CONFIG_ADT7310_TRIGGER_OWN_THREAD=y
```

Comment:
`CONFIG_ADT7310_TRIGGER=y` shouldn't be needed.

Move `CONFIG_ADT7310_TRIGGER_OWN_THREAD` up to preserve ordering

Please also add this sensor to `sensors_trigger_global.conf` and `sensors_trigger_none.conf`. See #56695